embedded/object/document.go

```diff
@@ -0,0 +1,114 @@
+package object
```

Comment:
probably package name should be `document`?

Reply:
I don't have any strong opinions on this, changing it to document